SU-HSDO/suhumsci
# STN-210: Main Navigation Styles

# READY FOR REVIEW

## Summary
This pull request adds styles for the main navigation. The mobile nav and secondary navigation share a lot of similarities however this pull request does not refactor the secondary navigation to ensure consistency. There are mixins and variables that can be used for a future refactor when the time comes.

This PR also includes the start of an icon system. Future SVGs should cleaned up and be saved as variables in `settings/_variables.icons.scss` to ensure consistency across the site.

This PR has been reviewed in the following browsers:
- Chrome (Window & Mac)
- Firefox (Window & Mac)
- Safari
- Edge
- IE 11 (JS does not work at the moment but the styles were still reviewed)

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. Run `npm start` to build new assets
2. Run `npm test` to check for errors
3. Confirm that the navigation matches the designs in Figma
    - The entire navigation should be accessible at small-medium screens in the main nav and the secondary nav should be hidden
    - Only the first 2 levels of navigation should accessible at large screens and the secondary nav should be visible
4. Play around with long lines of text in different at different levels
5. Confirm that the navigation can have dropdowns and pages at first level (pages will never have dropdowns)

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
